source.lua

```diff
@@ -0,0 +1,145 @@
+--@ module = true
+-- wip
+
+local repeatUtil = require('repeat-util')
+
+liquidSources = liquidSources or {}
+
+local sourceId = 'liquidSources'
+
+function isFlowPassable(pos)
+    local tiletype = dfhack.maps.getTileType(pos.x, pos.y, pos.z)
+    local titletypeAttrs = df.tiletype.attrs[tiletype]
+    local shape = titletypeAttrs.shape
+    local tiletypeShapeAttrs = df.tiletype_shape.attrs[shape]
+    return tiletypeShapeAttrs.passable_flow
+end
+
+function AddLiquidSource(pos, liquid, amount)
+    table.insert(liquidSources, {
+        liquid = liquid,
+        amount = amount,
+        pos = {
+            x = pos.x,
+            y = pos.y,
+            z = pos.z,
+        },
+    })
+
+    repeatUtil.scheduleEvery(sourceId, 12, 'ticks', function()
+        if next(liquidSources) == nil then
+            repeatUtil.cancel(sourceId)
+        else
+            for _, v in pairs(liquidSources) do
+                local block = dfhack.maps.getTileBlock(v.pos.x, v.pos.y, v.pos.z)
```

Comment:
This wasn't working right when I tried, I *thought* it might have been that the single arg version of these functions required a `pos` pointer but considering the suggestion I'll have to try again, I was likely doing something else wrong.

Reply:
I realize now I got it backwards, I had a pointer, it was expecting a table. An earlier version was mispassing the argument instead of the iterated value, which was in fact a pointer. Compounding problem yielding a confusing result, the classic.